src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,774 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+'''
+Classes for estimators for quantum phase estimation. Estimators described
+in arXiv:XXXX.XXXX
+'''
+
+import numpy as np
+from numpy import cos, sin, pi
+from scipy import sparse, optimize
+import warnings
+from openfermion.config import EQ_TOLERANCE
+
+
+class ProbabilityDist(object):
+    '''
+    Stores a estimate of multiple phases
+    $P(\phi_j)=p_0+\sum_{r=0}^N (p_{2j}cos(r*\phi_j)+p_{2j-1}sin(r*\phi_j))$
+    and the probability $A_j=|a_j|^2$, with $a_j$ the amplitude of the
+    eigenstate $|\phi_j\>$ in the starting state of the experiment.
+    '''
+
+    def __init__(self,
+                 amplitude_guess=[1],
+                 amplitude_vars=[[1]],
+                 num_vectors=1,
+                 num_freqs=1000,
+                 max_n=1,
+                 vector_guess=None):
+
+        '''
+        Args:
+            amplitude_guess: estimates for amplitudes of different
+                eigenstates in the initial state ($A_j$)
+            amplitude_vars: variance on the amplitude estimates
+            num_vectors: number of phases $\phi_j$ to estimate
+            num_freqs: number of frequencies in Fourier representation.
+            max_n: maximum number of unitary evolutions before measurement.
+                dictates how many matrices for updates are made and stored.
+            vector_guess: a prior estimate of the phases $\phi_j$. If none,
+                assumes flat.
+        '''
+
+        # Store size data
+        self._num_vectors = num_vectors
+        self._num_freqs = num_freqs
+
+        # Store prior information
+        if vector_guess is not None and (
+                num_vectors != vector_guess.shape[1]
+                or 2*num_freqs+1 != vector_guess.shape[0]):
+            raise ValueError('''Prior distribution of phases has shape {},
```

Comment:
please add
`Raises:
    ValueError: Prior distribution...`
to the doc strings (in the same style you do for 'Args')

Reply:
Done